packages/waku/src/lib/plugins/vite-plugin-fs-router-typegen.ts

```diff
@@ -91,20 +71,26 @@ export const fsRouterTypegenPlugin = (opts: { srcDir: string }): Plugin => {
       }
 
       // Recursively collect `.tsx` files in the given directory
-      const collectFiles = async (dir: string): Promise<string[]> => {
-        if (!pagesDir) return [];
-        const results: string[] = [];
-        const files = await readdir(dir, {
-          withFileTypes: true,
-          recursive: true,
-        });
-
-        for (const file of files) {
-          if (file.name.endsWith('.tsx')) {
-            results.push('/' + file.name);
+      const collectFiles = async (
+        dir: string,
+        cwd: string = '',
+        files: string[] = [],
+      ): Promise<string[]> => {
+        if (!cwd) {
+          cwd = dir;
+        }
+        const entries = await readdir(cwd, { withFileTypes: true });
+        for (const entry of entries) {
+          const fullPath = joinPath(cwd, entry.name);
+          if (entry.isDirectory()) {
+            await collectFiles(dir, fullPath, files);
+          } else {
+            if (entry.name.endsWith('.tsx')) {
+              files.push(fullPath.slice(dir.length));
+            }
           }
         }
-        return results;
+        return files;
       };
 
       const fileExportsGetConfig = (filePath: string) => {
```

Comment:
Should filePath be renamed? or is this still right?

Reply:
filePath seems right.